web/src/js/ducks/settings.js

```diff
@@ -0,0 +1,92 @@
+import { StoreCmds } from '../actions'
+import { addLogEntry } from './eventLog'
+
+export const WS_MSG_TYPE = 'settings'
+export const WS_MSG_CMD_RESET = 'reset'
+export const WS_MSG_CMD_UPDATE = 'update'
+
+export const BEGIN_FETCH = 'SETTINGS_BEGIN_FETCH'
+export const FETCHED = 'SETTINGS_FETCHED'
+export const RESET = 'SETTINGS_RESET'
+export const FETCH_ERROR = 'SETTINGS_FETCH_ERROR'
+export const RECV_WS_MSG = 'SETTINGS_RECV_WS_MSG'
+
+const defaultState = { settings: {}, pendings: null, req: null }
+
+export default function reduce(state = defaultState, action) {
+    switch (action.type) {
+
+        case BEGIN_FETCH:
+            return { ...state, pendings: [], req: action.req }
+
+        case FETCHED:
+            const pendings = state.pendings || []
+            return { ...state, pendings: null, settings: pendings.reduce(reduceData, action.data) }
+
+        case RESET:
+            return { ...state, pendings: null, settings: action.data || {} }
+
+        case RECV_WS_MSG:
+            if (state.pendings) {
+                return { ...state, pendings: state.pendings.concat(action) }
+            }
+            return { ...state, settings: reduceData(state.settings, action) }
+
+        default:
+            return state
+    }
+}
+
+function reduceData(data, action) {
+    switch (action.cmd) {
+
+        case WS_MSG_CMD_RESET:
+            return action.data || {}
+
+        case WS_MSG_CMD_UPDATE:
+            return _.merge({}, data, action.data)
+
+        default:
+            return data
+    }
+}
+
+export function fetch() {
+    return dispatch => {
+        const req = $.getJSON('/settings')
```

Comment:
@mhils I'd love to, but fetch API is not cancelable so we'll need to have a workaround for that